Re-enable package validation and clean up package content

ericstj opened the pull request:
Requires: https://github.com/dotnet/buildtools/pull/1441

This re-enables package validation and fixes all validation issues with packages.

I've deleted packages for System.Net.Http and System.Runtime.InteropServices.RuntimeInformation which were absorbed by NETStandard.
I've deleted packages for System.Runtime.WindowsRuntime.* since these are only supported by UWP and are in the framework package.

Harvesting is now explicit so we'll no longer see "magic" happening around assets being automatically redistributed.

This will fail CI until we have my buildtools change.

safern commented:
test Innerloop CentOS7.1 Debug x64 Build and Test (#18431)
test Innerloop PortableLinux Release x64 Build and Test (#18431)

weshaggard commented on external/netstandard/netstandard.depproj:
```diff
@@ -28,6 +29,12 @@
     <PackageReference Include="System.Runtime.Serialization.Primitives">
       <Version>4.3.0</Version>
     </PackageReference>
+    <PackageReference Condition="'$(NETStandardVersion)' &gt;= 1.3" Include="System.Reflection.TypeExtensions">
```
Does this condition actually work? Does msbuild call parse on the string?

ericstj replied:
MSBuild can handle numeric comparisons on integers and decimals. https://msdn.microsoft.com/en-us/library/7szfhaft.aspx

weshaggard commented on src/Microsoft.CSharp/pkg/Microsoft.CSharp.pkgproj:
```diff
@@ -31,6 +33,13 @@
+    <HarvestIncludePaths Include="ref/netstandard1.0" />
+    <HarvestIncludePaths Include="lib/netstandard1.3" />
+    <ValidatePackageSuppression Include="PermitMissingImplementation">
+      <!-- Supported inbox by all ns1.0-1.2 frameworks -->
```
Is there a reason that the InboxOnTargetFramework metadata doesn't help automate this?

ericstj replied:
This case is rare enough that I wanted to make it a suppression.  Ideally we'd actually have a matching impl and let inbox override that.

weshaggard commented on src/Microsoft.CSharp/ref/Microsoft.CSharp.csproj:
```diff
@@ -2,6 +2,8 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
+    <!-- must match version supported by frameworks which support 4.0.* inbox -->
+    <AssemblyVersion>4.0.0.0</AssemblyVersion>
```
If we do add more APIs then should we jump to whatever version is in the dir.props and harvest the 4.0.0.0 version?

ericstj replied:
Yes

ericstj replied:
I added a comment

weshaggard commented on src/Microsoft.Win32.Registry/pkg/Microsoft.Win32.Registry.pkgproj:
```diff
@@ -3,12 +3,18 @@
   <ItemGroup>
     <ProjectReference Include="..\ref\Microsoft.Win32.Registry.csproj">
-      <SupportedFramework>net461;netcoreapp2.0</SupportedFramework>
+      <SupportedFramework>net461;netcoreapp2.0;uap10.1;$(AllXamarinFrameworks)</SupportedFramework>
```
uap support just got removed for this library https://github.com/dotnet/corefx/pull/18507

ericstj replied:
Yeah, I dealt with that in the rebase.  UAP will still be supported, only now with a PlatformNotSupported assembly.

ericstj replied:
Fixing up package validation for this after rebase.

weshaggard commented on src/System.Data.Odbc/src/System.Data.Odbc.csproj:
```diff
@@ -117,7 +117,7 @@
     <Reference Include="System.Data" />
   </ItemGroup>
-  <ItemGroup>
+  <ItemGroup Condition="'$(IsPartialFacadeAssembly)' == 'true'">
```
Why would you want these references for netfx?

ericstj replied:
That was just a typo, I've already fixed it.

weshaggard commented on src/System.Data.SqlClient/src/System.Data.SqlClient.csproj:
```diff
@@ -7,13 +7,27 @@
     <!-- Although we have a NS configuration, we know we use API that's not currently whitelisted: just validate against OneCore -->
     <UWPCompatible>false</UWPCompatible>
+    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'netfx'">true</IsPartialFacadeAssembly>
+    <GeneratePlatformNotSupportedAssembly Condition="'$(TargetsWindows)' != 'true' AND '$(TargetsUnix)' != 'true'">true</GeneratePlatformNotSupportedAssembly>
```
I assume this condition is about getting all the configurations where you didn't specify an OS?

ericstj replied:
Yep

ericstj replied:
Would it be clearer to say OSGroup==AnyOS?

weshaggard replied:
I think it might be a little clearer to me yes.

ericstj replied:
OK

weshaggard commented on src/System.IO.Pipes/tests/Performance/System.IO.Pipes.Performance.Tests.csproj:
```diff
@@ -7,9 +7,9 @@
   <!-- Default configurations to help VS understand the configurations -->
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' " />
   <!-- Compiled Source Files -->
```
rogue comment.

ericstj replied:
Fixed

weshaggard commented on src/System.Net.Http.Rtc/pkg/System.Net.Http.Rtc.pkgproj:
```diff
@@ -1,13 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
I assume you plan to dead in this package? If so should we just delete the entire library?

ericstj replied:
We need the lib for binary compatibility with previous PCLs which UAP supported.

weshaggard replied:
I guess that is fine. We should consider harvesting things like this.

ericstj replied:
I don't think we should set up a case where we intentionally hinder our ability to produce an active shipping binary on one of our supported platforms so I'd lean towards not harvesting it.

weshaggard commented on src/System.Net.Http/pkg/System.Net.Http.pkgproj:
```diff
@@ -1,42 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Wow we can get away with eliminating this package completely?

ericstj replied:
Yes.  It is part of NETStandard.Library.  The only thing the package was kept around for was to deliver the OOB impl to desktop.  Now that is part of the desktop support package.

weshaggard commented on src/System.Reflection.DispatchProxy/ref/System.Reflection.DispatchProxy.csproj:
```diff
@@ -3,13 +3,11 @@
     <ProjectGuid>{7DF3C428-AAD6-41C7-98E6-6CACFD5C391E}</ProjectGuid>
-    <!-- UAP10.1 is not yet mapped to netstandard2.0, manually duplicate this ref -->
-    <PackageTargetFramework>netstandard2.0;uap10.1</PackageTargetFramework>
+    <PackageTargetFramework>netstandard2.0</PackageTargetFramework>
```
Why do we need to list the PackageTargetFramework explicitly in this guy now?

ericstj replied:
I can remove it.

weshaggard commented on src/System.Reflection.TypeExtensions/pkg/System.Reflection.TypeExtensions.pkgproj:
```diff
@@ -6,19 +6,27 @@
       <SupportedFramework>net461;netcoreapp2.0;$(AllXamarinFrameworks)</SupportedFramework>
     </ProjectReference>
     <ProjectReference Include="..\src\System.Reflection.TypeExtensions.csproj" />
+    <HarvestIncludePaths Include="ref/net46;lib/net46" />
```
nit: Some projects you list the InboxOnTargetFramework's before HarvestIncludePaths and others you do the reverse.

ericstj replied:
I'll try to make it consistent.

ericstj commented:
@weshaggard I addressed your feedback with minor ammends where appropriate.  Can you please review these new commits:
https://github.com/dotnet/corefx/pull/18498/commits/d7241ce8cd613d90b51e97e35b423ad5978d0d0a
https://github.com/dotnet/corefx/pull/18498/commits/e7533ab2c24728a5937bfb68609e183833359b27
https://github.com/dotnet/corefx/pull/18498/commits/722f331f19ee1a72fdee8e2c667e8aa5b9c0fc27

weshaggard commented on src/Microsoft.Win32.Registry.AccessControl/pkg/Microsoft.Win32.Registry.AccessControl.pkgproj:
```diff
@@ -6,11 +6,11 @@
       <SupportedFramework>net461;netcoreapp2.0;$(AllXamarinFrameworks)</SupportedFramework>
     </ProjectReference>
     <ProjectReference Include="..\src\Microsoft.Win32.Registry.AccessControl.csproj" />
-    <NotSupportedOnTargetFramework Include="netcore50">
```
Just so I'm clear your goal is now that anytime we would have done this in the past we would now put a GeneratePlatformNotSupported assembly in its place?

ericstj replied:
Yes, either that or let the win32 one that fails the WACK to install.

ericstj commented:
@dotnet-bot test Innerloop CentOS7.1 Debug x64 Build and Test
@dotnet-bot test Innerloop Ubuntu14.04 Release x64 Build and Test

ericstj commented:
Actually the failures here look like they are caused by my change:
```
/mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/pkg/frameworkPackage.targets(108,5): error : netstandard > System.IO.IsolatedStorage > System.Security.AccessControl > netstandard 
```

I changed System.Security.AccessControl to build against netstandard.  It's a PlatformNotSupported assembly on Unix.  That raises the question of wether or not our unix build of IsolatedStorage even works correctly...  /cc @JeremyKuhne

ericstj commented:
Wow, that isn't good.  Somehow on rerun Jenkins decided to ignore the failure:
```
15:32:52 /mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/pkg/frameworkPackage.targets(108,5): error : Cycle detected for /mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/bin/obj/shims/netcoreapp/facades/netstandard.dll. [/mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
15:32:52 /mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/pkg/frameworkPackage.targets(108,5): error : netstandard > System.IO.IsolatedStorage > System.Security.AccessControl > netstandard [/mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
15:32:52     0 Warning(s)
15:32:52     2 Error(s)
15:32:52 
15:32:52 Time Elapsed 00:12:01.69
15:32:53 Command execution succeeded.
```

@weshaggard any idea what's going on there?

weshaggard commented:
>@weshaggard any idea what's going on there?

No I'm not sure what is going on there. It looks like we perhaps the return code might be getting lost on the xplat msbuild.

weshaggard commented:
I filed https://github.com/dotnet/corefx/issues/18580 to look into the build failure being ignored.

ericstj commented:
Fixes #17914

JeremyKuhne commented:
> That raises the question of wether or not our unix build of IsolatedStorage even works correctly... /cc @JeremyKuhne

AccessControl isn't actually used when running on Unix- we do a runtime check.
